File: scripts/seed.js
const { db } = require('@vercel/postgres');
const { points_seed } = require('../app/lib/placeholder-data.js');
const bcrypt = require('bcrypt');

// async function seedCards(client) {
//   try {
//     // Create the "users" table if it doesn't exist
//     const createTable = await client.sql`
//     CREATE TABLE cards (
//       id SERIAL PRIMARY KEY,
//       name VARCHAR(255) UNIQUE
//     );
//   `;

//     console.log(`Created "users" table`);

//     // Insert data into the "users" table
//     const insertedCards = await Promise.all(
//       cards.map(async (card) => {
//         return client.sql`
//         INSERT INTO Cards (name)
//         VALUES (${card.name})
//         ON CONFLICT DO NOTHING;
//       `;
//       }),
//     );

//     console.log(`Seeded ${insertedCards.length} cards`);

//     return {
//       createTable,
//       users: insertedCards,
//     };
//   } catch (error) {
//     console.error('Error seeding cards:', error);
//     throw error;
//   }
// }

async function seedPoints(client) {
  try {
    await client.sql`DROP TABLE IF EXISTS points;`;
    // Create the "points" table if it doesn't exist
    const createTable = await client.sql`
    CREATE TABLE IF NOT EXISTS points (
      id SERIAL PRIMARY KEY,
      card_name VARCHAR(255) UNIQUE,
      points DECIMAL(10, 2),    
      week_start_date DATE,
      UNIQUE (card_name, week_start_date)
    );
  `;

    console.log(`Created "points" table`);

    // Insert data into the "invoices" table
    const insertedPoints = await Promise.all(
      points_seed.map(
        (points) => client.sql`
        INSERT INTO points (card_name, points, week_start_date)
        VALUES (${points.card_name}, ${points.points}, ${points.week_start_date})
        ON CONFLICT (card_name, week_start_date)
        DO UPDATE SET card_name = excluded.card_name, points = excluded.points, week_start_date = excluded.week_start_date;
      `,
      ),
    );

    console.log(`Seeded ${insertedPoints.length} points`);

    return {
      createTable,
      points: insertedPoints,
    };
  } catch (error) {
    console.error('Error seeding points:', error);
    throw error;
  }
}

async function main() {
  const client = await db.connect();

  await seedPoints(client);
  await client.end();
}

main().catch((err) => {
  console.error(
    'An error occurred while attempting to seed the database:',
    err,
  );
});

File: scripts/seed.py
import psycopg2
import os
from psycopg2 import sql
from psycopg2.extras import execute_values
from placeholder_data import points_seed
from dotenv import load_dotenv

load_dotenv()  # take environment variables from .env.
def seed_points(conn):
    try:
        with conn.cursor() as cur:
            # Drop the "points" table if it exists
            cur.execute("DROP TABLE IF EXISTS points;")
            
            # Create the "points" table if it doesn't exist
            create_table_query = """
            CREATE TABLE IF NOT EXISTS points (
                id SERIAL PRIMARY KEY,
                card_name VARCHAR(255),
                points DECIMAL(10, 2),    
                week INTEGER,
                UNIQUE (card_name, week)
            );
            """
            cur.execute(create_table_query)
            print("Created 'points' table")

            # Insert data into the "points" table
            insert_query = """
            INSERT INTO points (card_name, points, week)
            VALUES %s
            ON CONFLICT (card_name, week)
            DO UPDATE SET card_name = excluded.card_name, points = excluded.points, week = excluded.week;
            """
            data_tuples = [(point['card_name'], point['points'], point['week']) for point in points_seed]
            execute_values(cur, insert_query, data_tuples)
            print(f"Seeded {len(data_tuples)} points")

        conn.commit()
    except Exception as error:
        print('Error seeding points:', error)
        raise

def main():
    print("os.getenv('POSTGRES_URL')", os.getenv("POSTGRES_URL"))
    conn = psycopg2.connect(os.getenv("POSTGRES_URL"))
    seed_points(conn)
    conn.close()

if __name__ == "__main__":
    main()